### DISCUSSION TITLE
How to do a SCAN?

### DISCUSSION THREAD

**Discussion by JeffBNimble** [redis/rueidis#348] [Q&A] [answered]
So, I've read the Common Examples section of the README and there is no example of doing a SCAN.

This is the best I've come up with, but unsure if rueidis does the iteration by cursor for me or do I have to?

`
cursor := 0
count := 100
pattern := "someKeyPrefix:*"
client.Do(client.B().Scan().Cursor(cursor).Match(pattern).Count(count).Type(?).Build())
`
Here are my questions:
1. I'm confused by Scan Type. What do I specify for that? 
2. How do I get the keys back that match the pattern?
3. Am I able to use Scan with DoCache()? If so, what does that end up doing? Does it just scan the local cache?

**Comment by rueian** [comment, marked as answer]
Hi @JeffBNimble,

Rueidis doesn't do the iteration. How to do the iteration is up to you. Here is an [example](https://pkg.go.dev/github.com/redis/rueidis#example-Client-Scan) for you to reference:

https://github.com/redis/rueidis/blob/6556f2fa4d07c416b3029f04419ef77067f09b8a/rueidis_test.go#L369-L385

1. The `Type` parameter is optional. It serves as a data type filter. You can find more details [here](https://github.com/redis/redis/pull/6116).
2. As shown in the above example, you can use the `AsScanEntry()` helper and get the keys from the `scan.Elements`.
3. No, `DoCache()` only works with read-only commands having key specified.

**Reply by JeffBNimble** [reply]
Thank you!
